sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/ResolveSubquerySuite.scala

```diff
@@ -177,4 +178,61 @@ class ResolveSubquerySuite extends AnalysisTest {
       condition = Some(sum('a) === sum('c)))
     assertAnalysisError(plan, Seq("Invalid expressions: [sum(a), sum(c)]"))
   }
+
+  test("SPARK-35618: lateral join with star expansion") {
```

Comment:
Could you add tests for case-sensitivities?

Reply:
Also, I think we need some tests for `spark.sql.parser.quotedRegexColumnNames`.

Reply:
@maropu I looked into how regex expressions are resolved and the logic is actually different from star expressions. It won't throw exceptions when there is no match. Instead, it returns an empty sequence. So we can't tell if the regex expression is resolved by the current plan with an empty output, or it can't be resolved.